hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -141,7 +154,7 @@ public synchronized int read(final byte[] b, final int off, final int len) throw
     return totalReadBytes > 0 ? totalReadBytes : lastReadBytes;
   }
 
-  private int readOneBlock(final byte[] b, final int off, final int len) throws IOException {
+  private int validate(byte[] b, int off, int len) throws IOException {
```

Comment:
That would alter the existing flow. In certain cases it has to return 0 and -1 in some other cases, exceptions in case validations failed for len, off values.

Reply:
Returning int from validate which is used by caller as bytes read is wrong. Any failure in validate should return Boolean false and the caller in turn can convert it to -1 for bytes read.

Reply:
Done